presto-main/src/main/java/com/facebook/presto/metadata/FunctionUtils.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.metadata;
+
+import com.facebook.presto.spi.function.OperatorType;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.spi.type.TypeSignature;
+import com.facebook.presto.spi.type.VarcharType;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.primitives.Primitives;
+import io.airlift.slice.Slice;
+
+import java.util.Set;
+
+import static com.facebook.presto.metadata.FunctionKind.SCALAR;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.spi.type.VarbinaryType.VARBINARY;
+import static com.google.common.base.Preconditions.checkArgument;
+
+public final class FunctionUtils
```

Comment:
I feel like this should have a more specific name like: `FunctionSignatureUtils` or `SignatureMangler`.... not sureI like those two, but something more specific than "utilities for functions".

Reply:
This is one of the commits that's more or less still kept its look from the original PR. I think the intention is to move the static methods out so other classes don't reference `FunctionRegistry` directly. However, if you look at the methods here, only `mangleOperatorName` is widely used. All others can potentially be moved somewhere else. I was debating whether I want to do that now, but figured that maybe it's better to do it later once `FunctionRegistry` is removed and the logic for operators and functions are sorted out into proper places.

Reply:
If we can make the literal changes I suggested, I think we can rename this to something like `OperatorSignatureUtils`